xds/src/test/java/io/grpc/xds/FallbackLbTest.java

```diff
@@ -0,0 +1,377 @@
+      balancers1.add(balancer);
+      return balancer;
+    }
+  };
+
+  private final LoadBalancerProvider fallbackProvider2 = new LoadBalancerProvider() {
+    @Override
+    public boolean isAvailable() {
+      return true;
+    }
+
+    @Override
+    public int getPriority() {
+      return 5;
+    }
+
+    @Override
+    public String getPolicyName() {
+      return "fallback_2";
+    }
+
+    @Override
+    public LoadBalancer newLoadBalancer(Helper helper) {
+      // just return mock and recored helper and balancer
+      helpers2.add(helper);
+      LoadBalancer balancer = mock(LoadBalancer.class);
+      balancers2.add(balancer);
+      return balancer;
+    }
+  };
+
+  private final Helper helper = mock(Helper.class);
+  private final List<Helper> helpers1 = new ArrayList<>();
+  private final List<Helper> helpers2 = new ArrayList<>();
+  private final List<LoadBalancer> balancers1 = new ArrayList<>();
+  private final List<LoadBalancer> balancers2 = new ArrayList<>();
+
+  private LoadBalancer fallbackLb;
+
+  @Before
+  public void setUp() {
+    LoadBalancerRegistry lbRegistry = new LoadBalancerRegistry();
+    lbRegistry.register(fallbackProvider1);
+    lbRegistry.register(fallbackProvider2);
+    fallbackLb = new FallbackLb(helper, lbRegistry);
+
+    assertThat(helpers1).isEmpty();
+    assertThat(helpers2).isEmpty();
+    assertThat(balancers1).isEmpty();
+    assertThat(balancers2).isEmpty();
+  }
+
+  @Test
+  public void handlePolicyChanges() throws Exception {
+    EquivalentAddressGroup eag111 = new EquivalentAddressGroup(mock(SocketAddress.class));
+    EquivalentAddressGroup eag112 = new EquivalentAddressGroup(mock(SocketAddress.class));
+    List<EquivalentAddressGroup> eags11 = ImmutableList.of(eag111, eag112);
+    String lbConfigRaw11 = "{"
+        + "\"balancerName\" : \"dns:///balancer.example.com:8080\","
+        + "\"fallbackPolicy\" : [{\"fallback_1\" : { \"fallback_1_option\" : \"yes\"}}]"
+        + "}";
+    @SuppressWarnings("unchecked")
+    Map<String, ?> lbConfig11 = (Map<String, ?>) JsonParser.parse(lbConfigRaw11);
+    fallbackLb.handleResolvedAddresses(ResolvedAddresses.newBuilder()
+        .setAddresses(eags11)
+        .setAttributes(Attributes.newBuilder().set(ATTR_LOAD_BALANCING_CONFIG, lbConfig11).build())
+        .build());
+
+    assertThat(helpers1).hasSize(1);
+    assertThat(balancers1).hasSize(1);
+    Helper helper1 = helpers1.get(0);
+    LoadBalancer balancer1 = balancers1.get(0);
+    verify(balancer1).handleResolvedAddresses(ResolvedAddresses.newBuilder()
+        .setAddresses(eags11)
+        .setAttributes(Attributes.newBuilder()
+            .set(ATTR_LOAD_BALANCING_CONFIG, ImmutableMap.of("fallback_1_option", "yes")).build())
+        .build());
+
+    SubchannelPicker picker1 = mock(SubchannelPicker.class);
+    helper1.updateBalancingState(CONNECTING, picker1);
+    verify(helper).updateBalancingState(CONNECTING, picker1);
```

Comment:
This is verifying `helper1` is delegating to `helper` before policy name change, in comparison with `helper1` is no longer delegating after policy name change.

Reply:
This does nothing. Of course `helper1` wraps `helper`. Putting this two lines together does not verify anything other than "`helper1` is a forwarder of `helper`", which is what `ForwardingLoadBalancerHelperTest` should do.

Reply:
It may not be a `ForwardingHelper`, even it is, it's all implementation detail. If it is simply a `ForwardingHelper`, how could the later verification
```
 helper1.updateBalancingState(CONNECTING, picker1);
 verify(helper, never()).updateBalancingState(CONNECTING, picker1);
```

pass?

Reply:
Fine.